### PR TITLE
change some factory methods

### PR DESCRIPTION
**Pull request by LipuFei**
Change some factory methods

### CODE REVIEW

File: community.py [hunk 1]
```diff
@@ -166,15 +166,21 @@ def get_master_members(cls, dispersy):
...
         """
-        Loading, or joining an existing community can be done by calling the constructor.
-        Creating a new instance of a community needs to use the create_community() call.
+        Create a new community owned by my_member.
+
+        Each unique community, that exists out in the world, is identified by a public/private key
+        pair.  When the create_community method is called such a key pair is generated.
+
+        Furthermore, my_member will be granted permission to use all the messages that the community
+        provides.
```
Comment (NielsZeilemaker) [inline community.py]: This text needs to change, lets do something like:

```
Instantiate a new community, using master as the identifier and my_member as the public/private keypair to be used when sending messages. 
Each community is identified by the hash of the public key of the master member. This member is created in the create_community method.
```

File: community.py [hunk 2]
```diff
@@ -182,9 +188,33 @@ def __init__(self, dispersy, master, my_member, attach=True):
         @param master: The master member that identifies the community.
         @type master: DummyMember or Member
 
-        @param my_member: The my member that identifies you in this community.
+        @param my_member: The Member that will be granted Permit, Authorize, and Revoke for all
```
Comment (NielsZeilemaker) [inline community.py]: Not true, please revert

File: community.py [hunk 3]
```diff
@@ -182,9 +188,33 @@ def __init__(self, dispersy, master, my_member, attach=True):
...
+         messages.
         @type my_member: Member
+
+        @param args: optional arguments that are passed to the community constructor.
+        @type args: tuple
+
+        @param kargs: optional keyword arguments that are passed to the community constructor.
+        @type args: dictionary
```
Comment (NielsZeilemaker) [inline community.py]: kargs instead of args

File: community.py [hunk 4]
```diff
@@ -1802,8 +1855,22 @@ def get_member(self, *argv, **kwargs):
         member = self._dispersy.get_member(mid=mid, public_key=public_key, private_key=private_key)
         # We only need to check if this member has an identity message in this community if we still don't have the full
         # public key
-        if not mid or (member and member.has_identity(self)):
+        if not mid:
             return member
+        if member and isinstance(member, Member):
```
Comment (NielsZeilemaker) [inline community.py]: You could change this line to

```
if isinstance(member, Member):
```

**Comment by LipuFei** [comment]
@NielsZeilemaker fixed.

File: community.py [hunk 5]
```diff
@@ -166,15 +166,22 @@ def get_master_members(cls, dispersy):
...
         """
-        Loading, or joining an existing community can be done by calling the constructor.
-        Creating a new instance of a community needs to use the create_community() call.
+        Initializes a new community, using master as the identifier and my_member as the
+        public/private keypair to be used when sending messages.
+
+        Each community is identified by the hash of the public key of the master member.
+        This member is created in the create_community method.
+
+        Furthermore, my_member will be granted permission to use all the messages that the community
```
Comment (NielsZeilemaker) [inline community.py]: Drop this line, as its not true

**Comment by LipuFei** [comment]
@NielsZeilemaker comments updated.

**Comment by NielsZeilemaker** [comment]
Thanks, lipu i'll merge it when the pull request tester passes.

**Comment by LipuFei** [comment]
Then I'll continue the async db work :)   nice weekend.

**Pull request merged**